src/vsr/replica.zig

```diff
@@ -41,6 +41,24 @@ pub const Status = enum {
     recovering_head,
 };
 
+const CommitStatus = enum {
```

Comment:
I wonder if there's a better name, to reflect that we lineary move throughout the stages. `CommitProgress` or `CommitStage` perhaps? Or, to name the pattern "flow"

```
fn commit_flow(self: *Self, flow: CommitFlow) void {
```

Status also feels ok as well though.

Reply:
Good point! I went with `CommitStage`, and left the function name as-is (`commit_dispatch`).